lib/rspec/rails/matchers/have_enqueued_mail.rb

```diff
@@ -88,6 +88,18 @@ def arguments_match?(job)
           super(job)
         end
 
+        def process_arguments(job, given_mail_args)
+          if job[:job] == ActionMailer::MailDeliveryJob
+            if given_mail_args.first.is_a?(Hash) && job[:args][3]['params'].present?
```

Comment:
Is it necessarily a `[3]` or it may depend on something?

Reply:
I think is guaranteed that `[3]` always point to mail job args, especially when using one of builtin mailers:

In these example you have:

```
 :args=>["UnifiedMailer", "test_email", "deliver_now", {"params"=>{"foo"=>"bar", "_aj_symbol_keys"=>[]}, "args"=>[], "_aj_symbol_keys"=>["params", "args"]}],
```

the same assumption for args is present in the above code when building `base_mailer_args`